libcontainer/container_linux.go

```diff
@@ -201,18 +201,28 @@ func (c *Container) Set(config configs.Config) error {
 
 // Start starts a process inside the container. Returns error if process fails
 // to start. You can track process lifecycle with passed Process structure.
-func (c *Container) Start(process *Process) error {
+func (c *Container) Start(process *Process) (err error) {
 	c.m.Lock()
-	defer c.m.Unlock()
+	defer func() {
+		if err != nil {
```

Comment:
Can this Variable `err` be resolved here?

Reply:
Yes

Reply:
Are you ok with renaming these output variables (I tend to use `retErr` for these, but other names are fine); using the standard `err` for these can be confusing, and in some cases also a risk of accidentally shadowing.

Reply:
Yes, using `retErr` is more readable.

Reply:
Thanks! Yes, even if there's _technically_ not always a need; I like them, because when reading the code I'm immediately aware that there's "something special" with that error. And it prevents accidentally shadowing (as I mentioned above); shadowing definitely happened a few times, and can be really hard to spot if the function is more complex 👍